src/main/java/com/wolox/training/service/OpenLibraryService.java

```diff
@@ -0,0 +1,67 @@
+package com.wolox.training.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.wolox.training.exceptions.BookNotFoundException;
+import com.wolox.training.models.BookDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class OpenLibraryService {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${OpenLibraryUrl}")
+    String openLibraryUrl;
+
+    private BookDTO getBookNode(ObjectNode objectNode, String isbn){
+        final String isbnQueryValue = "ISBN:" + isbn;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        BookDTO bookDTO = new BookDTO();
+
+        bookDTO.setIsbn(isbn);
+        bookDTO.setTitle(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("title"), String.class));
+        bookDTO.setSubtitle(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("subtitle"), String.class));
+        bookDTO.setNumberOfPages(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("number_of_pages"), String.class));
+        bookDTO.setPublishDate(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("publish_date"), String.class));
+        bookDTO.setAuthors(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("authors"), List.class));
+        bookDTO.setPublisher(objectMapper.convertValue(objectNode.get(isbnQueryValue).get("publishers"), List.class));
```

Comment:
You can avoid setting these attributes by hand if you use the readvalue method of the ObjectMapper 😄  Check this out: https://www.baeldung.com/jackson-object-mapper-tutorial